### PR TITLE
Fix multivolume access tests for Windows

### PR DESCRIPTION
**Pull request by jingxu97**
Fix issue for multivolume access tests

Change-Id: I7e00d8bfe9682540b60b5998f78f2a48550934af

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/pod/create.go [hunk 1]
```diff
@@ -167,11 +166,12 @@ func MakeSecPod(podConfig *Config) (*v1.Pod, error) {
...
 		return nil, fmt.Errorf("Cannot create pod with empty namespace")
 	}
-	if len(podConfig.Command) == 0 {
+	if len(podConfig.Command) == 0 && !NodeOSDistroIs("windows") {
 		podConfig.Command = "trap exit TERM; while true; do sleep 1; done"
 	}
+
 	podName := "pod-" + string(uuid.NewUUID())
-	if podConfig.FsGroup == nil && runtime.GOOS != "windows" {
+	if podConfig.FsGroup == nil && !NodeOSDistroIs("windows") {
```
Comment (mauriciopoppe) [inline test/e2e/framework/pod/create.go:174]: `runtime.GOOS != "windows"` will always be true because we never run the e2e binary in a windows node

Comment (jingxu97) [reply test/e2e/framework/pod/create.go:174]: y, so here it fixed it, right?

Comment (mauriciopoppe) [reply test/e2e/framework/pod/create.go:174]: if it's always true then:

```
if podConfig.FsGroup == nil && runtime.GOOS != "windows" {
```

would be equal to

```
if podConfig.FsGroup == nil {
```

Comment (jingxu97) [reply test/e2e/framework/pod/create.go:174]: > if it's always true then:
> 
> ```
> if podConfig.FsGroup == nil && runtime.GOOS != "windows" {
> ```
> 
> would be equal to
> 
> ```
> if podConfig.FsGroup == nil {
> ```

But here we use NodeOSDistroIs instead runtime.GOOS?

Comment (mauriciopoppe) [reply test/e2e/framework/pod/create.go:174]: I understand now, thanks!

File: test/e2e/framework/pod/create.go [hunk 2]
```diff
@@ -167,11 +166,12 @@ func MakeSecPod(podConfig *Config) (*v1.Pod, error) {
 	if podConfig.NS == "" {
 		return nil, fmt.Errorf("Cannot create pod with empty namespace")
 	}
-	if len(podConfig.Command) == 0 {
+	if len(podConfig.Command) == 0 && !NodeOSDistroIs("windows") {
 		podConfig.Command = "trap exit TERM; while true; do sleep 1; done"
```
Comment (mauriciopoppe) [inline test/e2e/framework/pod/create.go:170]: with this other change:

```
	if command == "" {
		return commands
	}
```

is the pod still running in an endless loop?

Comment (jingxu97) [reply test/e2e/framework/pod/create.go:170]: I think so, without running any commands, agnhost container will keep running.

Comment (jingxu97) [reply test/e2e/framework/pod/create.go:170]: I also feel for most cases including linux, this command set up is not necessary. The test container most already does it.

Comment (mauriciopoppe) [reply test/e2e/framework/pod/create.go:170]: got it, thanks for sharing this, there was another place in the tests where we I saw sleep, I'm going to change that later

**Comment by mauriciopoppe** [comment]
/approve
/lgtm
/kind bug

I think you may need an approval from sig-testing too

**Pull request merged**